core/src/test/java/org/apache/druid/timeline/partition/SingleDimensionPartialShardSpecTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.timeline.partition;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class SingleDimensionPartialShardSpecTest
+{
+  @Test
+  public void testSerde() throws IOException
+  {
+    final SingleDimensionPartialShardSpec expected = new SingleDimensionPartialShardSpec(
+        "partitionKey",
+        3,
+        "start",
+        "end",
+        10
+    );
+    final ObjectMapper mapper = new ObjectMapper();
+    final byte[] json = mapper.writeValueAsBytes(expected);
+    final SingleDimensionPartialShardSpec fromJson = (SingleDimensionPartialShardSpec) mapper.readValue(
+        json,
+        PartialShardSpec.class
+    );
+    Assert.assertEquals(expected, fromJson);
+  }
+}
```

Comment:
May be useful to add an `EqualsVerifier` test to this class.

Reply:
Added.